### PR TITLE
Added article copy popup (used by enterprise app)

### PR DESCRIPTION
**Pull request by SamShiels**
Added the ability to copy permalink automatically with the permalink anchor. Note that this logic is overridden by the enterprise app due to base url conflicts. The functionality here is only used when running `hugo serve`.
Added styles for the link copy popup.

### Issue
https://spandigital.atlassian.net/browse/PRSDM-2965

### Screenshots
Screenshot 2022-12-08 at 17 15 48

### Checklist before merging

* [x] Did you test your changes locally?
* [x] Did you update the CHANGELOG?
* [ ] Is the documentation updated for this change? (If Required)

### CODE REVIEW

**Comment by SamShiels** [comment]
Drafted because I discovered a bug with the link copying.

**Review by CharlRitterCompany — APPROVED** [review]

**Comment by SamShiels** [comment]
Issues mentioned above have been resolved.

**Comment by Zalaras** [comment]
@SamShiels So it copies but I'm not getting a popup or icon to say that it has been copied

File: layouts/partials/article.html [hunk 1]
```diff
@@ -27,6 +27,17 @@
     {{ $content = len .Content }}
 {{ end }}
 
+<script>
+    function copyPermalink(link) {
+        // Copy the article link to the clipboard
+        let sanitizedLink = link;
+        while (sanitizedLink[0] === '/') {
+            sanitizedLink = sanitizedLink.substr(1);
```
Comment (effleurager) [inline layouts/partials/article.html:35]: [String.prototype.substr()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr) is deprecated

**Review by effleurager — CHANGES_REQUESTED** [review]

File: layouts/partials/article.html (hunk 1 above)
Comment (SamShiels) [reply layouts/partials/article.html]: Thanks! Changed it.

File: layouts/partials/article.html [hunk 2]
```diff
@@ -27,6 +27,17 @@
...
+<script>
+    function copyPermalink(link) {
+        // Copy the article link to the clipboard
+        let sanitizedLink = link;
+        while (sanitizedLink[0] === '/') {
+            sanitizedLink = sanitizedLink.substring(1);
+        }
+        navigator.clipboard.writeText(sanitizedLink);
+    }
+</script>
```
Comment (effleurager) [inline layouts/partials/article.html:39]: This whole script should be moved to the `baseof` layout, there's no sense in repeating it for each article

Comment (SamShiels) [reply layouts/partials/article.html]: Great point. Thanks. I moved it.

File: layouts/_default/baseof.html [hunk 3]
```diff
@@ -73,6 +73,17 @@ <h1 class="page-title">{{ .Title }}</h1>
...
     </script>
 
+    <script>
+      function copyPermalink(link) {
+          // Copy the article link to the clipboard
+          let sanitizedLink = link;
+          while (sanitizedLink[0] === '/') {
+              sanitizedLink = sanitizedLink.substring(1);
+          }
+          navigator.clipboard.writeText(sanitizedLink);
```
Comment (effleurager) [inline layouts/_default/baseof.html:83]: Last change, I promise 😂 
```suggestion
          navigator.clipboard.writeText(RegExp('^/*(.*)').exec(link)[1]);
```

**Review by effleurager — APPROVED** [review]

File: layouts/_default/baseof.html (hunk 3 above)
Comment (effleurager) [reply layouts/_default/baseof.html:83]: not a big fan of loops 💁🏽

Comment (SamShiels) [reply layouts/_default/baseof.html:83]: Thanks haha. This looks way better.

**Comment by SamShiels** [comment]
> @SamShiels So it copies but I'm not getting a popup or icon to say that it has been copied

@Zalaras That's only available on the React app. Since React is absent here.

**Pull request merged**